atlasdb-autobatch/src/main/java/com/palantir/atlasdb/autobatch/DisruptorAutobatcher.java

```diff
@@ -131,33 +134,55 @@ public BatchElement<T, R> consume() {
 
     public static final class DisruptorFuture<R> extends AbstractFuture<R> {
 
+        private final Ticker ticker;
         private final DetachedSpan parent;
-        private final DetachedSpan waitingSpan;
+        private final TimedDetachedSpan waitingSpan;
 
         @Nullable
-        private DetachedSpan runningSpan = null;
+        private TimedDetachedSpan runningSpan = null;
+
+        private volatile boolean valueSet = false;
 
-        public DisruptorFuture(String safeLoggablePurpose) {
-            this.parent = DetachedSpan.start(safeLoggablePurpose + " disruptor task");
-            this.waitingSpan = parent.childDetachedSpan("task waiting to be run");
+        @VisibleForTesting
+        DisruptorFuture(Ticker ticker, AutobatcherTelemetryComponents telemetryComponents) {
+            this.ticker = ticker;
+            this.parent = DetachedSpan.start(telemetryComponents.getSafeLoggablePurpose() + " disruptor task");
+            this.waitingSpan = TimedDetachedSpan.from(ticker, parent.childDetachedSpan("task waiting to be run"));
             this.addListener(
                     () -> {
                         waitingSpan.complete();
                         if (runningSpan != null) {
                             runningSpan.complete();
+                            parent.complete();
+
+                            if (valueSet) {
+                                telemetryComponents.markWaitingTimeAndRunningTimeMetrics(
```

Comment:
I should pass a `totalTime` derived from the parent span (as opposed to derived as the sum of running/waiting). This would help if one of the latter is messed (and/or in the case of gc pauses etc)